src/psyclone/f2pygen.py

```diff
@@ -919,55 +932,162 @@ def _check_initial_values(dtype, values):
+                .format(self.SUPPORTED_TYPES, datatype))
+
+        fort_fmt = FortranFormat(True, False)  # free form, strict
+        if dtype == "integer":
+            reader = FortranStringReader("integer :: vanilla")
+            reader.set_format(fort_fmt)
+            myline = reader.next()
+            self._decl = fparser1.typedecl_statements.Integer(parent.root,
+                                                              myline)
+        elif dtype == "real":
+            reader = FortranStringReader("real :: vanilla")
+            reader.set_format(fort_fmt)
+            myline = reader.next()
+            self._decl = fparser1.typedecl_statements.Real(parent.root, myline)
+        elif dtype == "logical":
+            reader = FortranStringReader("logical :: vanilla")
+            reader.set_format(fort_fmt)
+            myline = reader.next()
+            self._decl = fparser1.typedecl_statements.Logical(parent.root,
+                                                              myline)
+        else:
+            # Defensive programming in case SUPPORTED_TYPES is added to
+            # but not handled here
+            raise NotImplementedError(
+                "Internal error: type '{0}' is in DeclGen.SUPPORTED_TYPES "
+                "but not handled by constructor.".format(dtype))
+
+        # Add any kind-selector
+        if kind:
+            self._decl.selector = ('', kind)
+
+        super(DeclGen, self).__init__(parent=parent, datatype=datatype,
+                                      entity_decls=entity_decls,
+                                      intent=intent, pointer=pointer,
+                                      dimension=dimension,
+                                      allocatable=allocatable, save=save,
+                                      target=target,
+                                      initial_values=initial_values)
+
+
+class CharDeclGen(BaseDeclGen):
+    '''
+    Generates a Fortran declaration for character variables.
+
+    :param parent: node to which to add this declaration as a child
+    :type parent: :py:class:`psyclone.f2pygen.BaseGen`
+    :param list entity_decls: list of variable names to declare
+    :param str intent: the INTENT attribute of this declaration
+    :param bool pointer: whether or not this is a pointer declaration
+    :param str kind: the KIND attribute to use for this declaration
+    :param str dimension: the DIMENSION specifier (i.e. the xx in \
+                          DIMENSION(xx))
+    :param bool allocatable: whether this declaration is for an \
+                             ALLOCATABLE quantity
+    :param bool save: whether this declaration has the SAVE attribute
+    :param bool target: whether this declaration has the TARGET attribute
+    :param str length: expression to use for the (len=xx) selector
+    :param initial_values: Initial value to give each variable.
+    :type initial_values: list of str with same no. of elements as entity_decls
```

Comment:
Should it say here that this can be a string (with additional internal quotes) or a variable name here? Note, as mentioned before, I don't think a string should have to internally have quotes but I admit I can't see how to distinguish between a variable and a string in that case if we are just passing in a Python string.

Reply:
Good point. Done.